arrays/arrays.c

```diff
@@ -114,15 +160,29 @@ void arr_append(Array *arr, char *element) {
  *
  * Throw an error if the value is not found.
  *****/
+
+// do this later
+
 void arr_remove(Array *arr, char *element) {
 
   // Search for the first occurence of the element and remove it.
   // Don't forget to free its memory!
-
   // Shift over every element after the removed element to the left one position
-
   // Decrement count by 1
+  int i = 0;
+  // this is a flag to find the intrest of elements.
+  int found = -1; 
+
+
+  while (i < arr->count && found == -1)
```

Comment:
I'm not sure I understand this loop. There are a lot of "elegant" ways to implement the remove function, and I think that's what you're going for here, but I think for the first pass at this it would be better to use a more inefficient solution. You will probably want to loop through the array and free the string element the matches the one you're looking for. Then you want to loop through the array again and shift every element after the one you deleted to the left one spot.